src/controller/python/chip/clusters/Types.py

```diff
@@ -15,7 +15,10 @@
 #    limitations under the License.
 #
 
+from dataclasses import dataclass
 
+
+@dataclass(frozen=True)
```

Comment:
Yes, with this that's much better!  Now we guarantee that there is no state stored on these things, etc.